src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -213,5 +213,27 @@ private static MsalTokenResponse CreateMsalTokenResponseFromResult(string broker
 
             return MsalTokenResponse.CreateFromBrokerResponse(response);
         }
+
+        public async Task<IEnumerable<IAccount>> GetAccountsAsync(string clientID)
+        {
+            Dictionary<string, string> brokerPayload = new Dictionary<string, string>();
+            brokerPayload.Add(BrokerParameter.ClientId, clientID);
+
+            try
+            {
+                await _brokerHelper.InitiateBrokerHandshakeAsync(_activity).ConfigureAwait(false);
+
+                return _brokerHelper.GetBrokerAccountsInAccountManager(brokerPayload);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error("Broker Operation Failed to complete. In order to perform brokered authentication on android" +
+                    " you need to ensure that you have installed either Intune Company Portal (Version 5.0.4689.0 or greater) or Microsoft Authenticator (6.2001.0140 or greater).");
+                if (ex is MsalException)
+                    throw;
+                else
+                    throw new MsalClientException(MsalError.AndroidBrokerOperationFailed, ex.Message, ex);
```

Comment:
can you add logging through here? in which scenarios would an MsalException be thrown?

Reply:
The exception will be thrown if the broker handshake operation fails (this is already logged in a different method) or if Xamarin were to throw some exception from its code.

Reply:
We should never throw MsalException - only `MsalClientException` and `MsalServiceException`. `MsalException` exists only as a base class.